commit 39c7135a4d972d5f412b3288e4100851361d507c
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Fri Jun 2 16:44:39 2017 +0800

    Fix compatibility with Qt 5.9
    
    QFlags only works on Enums since Qt 5.9. This should not break old Qt
    versions.
    
    Change-Id: I80326bf39d5ebe8c20db19aac8ad8beaa3af7371

diff --git a/dde-desktop/model/dfileselectionmodel.cpp b/dde-desktop/model/dfileselectionmodel.cpp
index 296e0342..af588ddb 100644
--- a/dde-desktop/model/dfileselectionmodel.cpp
+++ b/dde-desktop/model/dfileselectionmodel.cpp
@@ -18,7 +18,7 @@ DFileSelectionModel::DFileSelectionModel(QAbstractItemModel *model, QObject *par
 
 bool DFileSelectionModel::isSelected(const QModelIndex &index) const
 {
-    if (m_currentCommand != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect))
+    if (m_currentCommand != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect))
         return QItemSelectionModel::isSelected(index);
 
     for (const QItemSelectionRange &range : m_selection) {
@@ -31,7 +31,7 @@ bool DFileSelectionModel::isSelected(const QModelIndex &index) const
 
 int DFileSelectionModel::selectedCount() const
 {
-    if (m_currentCommand != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect))
+    if (m_currentCommand != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect))
         return selectedIndexes().count();
 
     return m_lastSelectedIndex.row() - m_firstSelectedIndex.row() + 1;
@@ -40,7 +40,7 @@ int DFileSelectionModel::selectedCount() const
 QModelIndexList DFileSelectionModel::selectedIndexes() const
 {
     if (m_selectedList.isEmpty()) {
-        if (m_currentCommand != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect)) {
+        if (m_currentCommand != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect)) {
             m_selectedList = QItemSelectionModel::selectedIndexes();
         } else {
             for (const QItemSelectionRange &range : m_selection) {
@@ -57,7 +57,7 @@ void DFileSelectionModel::select(const QItemSelection &selection, QItemSelection
     if (!command.testFlag(NoUpdate))
         m_selectedList.clear();
 
-    if (command != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect)) {
+    if (command != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect)) {
         if (m_timer.isActive()) {
             m_timer.stop();
             updateSelecteds();
diff --git a/dde-file-manager-lib/models/dfileselectionmodel.cpp b/dde-file-manager-lib/models/dfileselectionmodel.cpp
index a61f3844..8650f66d 100644
--- a/dde-file-manager-lib/models/dfileselectionmodel.cpp
+++ b/dde-file-manager-lib/models/dfileselectionmodel.cpp
@@ -20,7 +20,7 @@ DFileSelectionModel::DFileSelectionModel(QAbstractItemModel *model, QObject *par
 
 bool DFileSelectionModel::isSelected(const QModelIndex &index) const
 {
-    if (m_currentCommand != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect))
+    if (m_currentCommand != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect))
         return QItemSelectionModel::isSelected(index);
 
     for (const QItemSelectionRange &range : m_selection) {
@@ -35,7 +35,7 @@ bool DFileSelectionModel::isSelected(const QModelIndex &index) const
 
 int DFileSelectionModel::selectedCount() const
 {
-    if (m_currentCommand != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect))
+    if (m_currentCommand != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect))
         return selectedIndexes().count();
 
     return m_lastSelectedIndex.row() - m_firstSelectedIndex.row() + 1;
@@ -44,7 +44,7 @@ int DFileSelectionModel::selectedCount() const
 QModelIndexList DFileSelectionModel::selectedIndexes() const
 {
     if (m_selectedList.isEmpty()) {
-        if (m_currentCommand != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect)) {
+        if (m_currentCommand != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect)) {
             m_selectedList = QItemSelectionModel::selectedIndexes();
         } else {
             for (const QItemSelectionRange &range : m_selection) {
@@ -61,7 +61,7 @@ void DFileSelectionModel::select(const QItemSelection &selection, QItemSelection
     if (!command.testFlag(NoUpdate))
         m_selectedList.clear();
 
-    if (command != QFlags<QItemSelectionModel::SelectionFlags>(Current|Rows|ClearAndSelect)) {
+    if (command != QItemSelectionModel::SelectionFlags(Current|Rows|ClearAndSelect)) {
         if (m_timer.isActive()) {
             m_timer.stop();
             updateSelecteds();
